### PR TITLE
Makefile: fix WHAT not overriding the default package path

### PR DESCRIPTION
**Pull request by sttts**
`WHAT=./foo... make ...` should replace the default path

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,12 +100,14 @@ imports: $(OPENSHIFT_GOIMPORTS)
 COUNT ?= 5
 
 .PHONY: test-e2e
+test-e2e: WHAT ?= ./test/e2e...
 test-e2e: install
-	go test -tags e2e -race -count $(COUNT) ./test/e2e... $(WHAT)
+	go test -tags e2e -race -count $(COUNT) $(WHAT)
 
 .PHONY: test
+test: WHAT ?= ./test/e2e...
```
Comment (stevekuznetsov) [inline Makefile:108]: This should be `./...`

File: Makefile [hunk 2]
```diff
@@ -100,12 +100,14 @@ imports: $(OPENSHIFT_GOIMPORTS)
 COUNT ?= 5
 
 .PHONY: test-e2e
+test-e2e: WHAT ?= ./test/e2e...
 test-e2e: install
-	go test -tags e2e -race -count $(COUNT) ./test/e2e... $(WHAT)
```
Comment (stevekuznetsov) [inline Makefile:104]: Someone will be very surprised when they choose something other than e2e here and the build tags mean their `make` target does nothing :)

File: Makefile (hunk 1 above)
Comment (sttts) [reply Makefile:109]: indeed

Comment (sttts) [reply Makefile]: fixed

File: Makefile (hunk 2 above)
Comment (sttts) [reply Makefile]: what do you suggest? Without this WHAT does not make sense either.

Comment (sttts) [reply Makefile]: where exactly do we use the tag today?

**Pull request merged**